src/Symfony/Component/Workflow/Tests/RegistryTest.php

```diff
@@ -36,6 +36,9 @@ protected function tearDown()
     public function testAddIsDeprecated()
     {
         $this->registry->add(new Workflow(new Definition(array(), array()), $this->getMockBuilder(MarkingStoreInterface::class)->getMock(), $this->getMockBuilder(EventDispatcherInterface::class)->getMock(), 'workflow1'), $this->createSupportStrategy(Subject1::class));
+
+        // the test simply ensures that the legacy code still works
+        $this->addToAssertionCount(1);
```

Comment:
a better way would be to assert that you can get it. Otherwise, keeping the method but without registering the workflow would pass the test while the BC layer would be broken.

Reply:
Good idea. Fixed